app/src/main/java/com/flexcode/yummy/presentation/meals_screen/MealsViewModel.kt

```diff
@@ -17,15 +13,15 @@ import kotlinx.coroutines.Job
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.launch
 
 @HiltViewModel
 class MealsViewModel @Inject constructor(
-    private val getMealsUseCase: GetMealsUseCase,
-    private val getCategoriesUseCase: GetCategoriesUseCase
+    private val container: UseCaseContainer
```

Comment:
this is great you can rename the container to usecaseContainer for easier understandability and readability

Reply:
I have refactored the parameter to `useCaseContainer `and have pushed the code. Let me know if that is satisfactory.